File: 104. Maximum Depth of Binary Tree/Solution.py
# LeetCode #104: Maximum Depth of Binary Tree

'''
Given a binary tree, find its maximum depth.

The maximum depth is the number of nodes along the longest path from the root node down to the farthest leaf node.
'''

# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def maxDepth(self, root):
        if root == None:
            return 0;
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
        

File: 104. Maximum Depth of Binary Tree/Solution.c
// LeetCode #104: Maximum Depth of Binary Tree

/*
Given a binary tree, find its maximum depth.

The maximum depth is the number of nodes along the longest path from the root node down to the farthest leaf node.
*/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int maxDepth(struct TreeNode* root) {
    if (root == NULL)
        return 0;
        
    int leftHeight = maxDepth(root->left);
    int rightHeight = maxDepth(root->right);
    
    if (leftHeight > rightHeight)
        return 1 + leftHeight;
    else
        return 1 + rightHeight;
}